### PR TITLE
Add alert-rules

### PR DESCRIPTION
**Pull request by StianHaugland1**

### CODE REVIEW

**Review by mtverraen — APPROVED** [review]

File: node-dns-exporter/templates/alert-rules.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ template "node-dns-exporter.name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    alertmanager: {{ .Values.alertManager | default "default" }}
```
Comment (Starefossen) [inline node-dns-exporter/templates/alert-rules.yaml:7]: Can you add "documentation" for `.Values.alertManager` in `values.yaml`?

File: node-dns-exporter/templates/alert-rules.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ template "node-dns-exporter.name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
```
Comment (Starefossen) [inline node-dns-exporter/templates/alert-rules.yaml:6]: Can you add the default values also (see statefullset.yaml)

File: node-dns-exporter/templates/alert-rules.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    alertmanager: {{ .Values.alertManager | default "default" }}
+spec:
+  groups:
+  - name: nodeDNS
```
Comment (Starefossen) [inline node-dns-exporter/templates/alert-rules.yaml:10]: `NodeDNS`

File: node-dns-exporter/values.yaml [hunk 4]
```diff
@@ -71,4 +71,16 @@ nodeDns:
   testHosts: "" #"example.com, youtube.com"
   
   #listenAddress: 0.0.0.0:8080
-  #testInterval: 120
\ No newline at end of file
+  #testInterval: 120
+
+alertFor: #3m  
+
+alertRules: []
```
Comment (Starefossen) [inline node-dns-exporter/values.yaml:78]: `customAlertRules`

File: node-dns-exporter/templates/alert-rules.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+apiVersion: monitoring.coreos.com/v1
```
Comment (Starefossen) [inline node-dns-exporter/templates/alert-rules.yaml:1]: Add an if switch, check if prometheus monitoring is enabled and add an option (default true) to enable alertmanager metrics.

**Comment by Starefossen** [comment]
Let's start here. But next version can set up the individual alerts for each test host as well as search domain to look for, but let's start here.

**Review by Starefossen — APPROVED** [review]

**Comment by Starefossen** [comment]
Looks good 🚀

**Pull request merged**